components/brave_sync/bookmark_model_loaded_only_observer.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SYNC_BOOKMARK_MODEL_LOADED_ONLY_OBSERVER_H_
+#define BRAVE_COMPONENTS_BRAVE_SYNC_BOOKMARK_MODEL_LOADED_ONLY_OBSERVER_H_
+
+#include <set>
+
+#include "components/bookmarks/browser/bookmark_model_observer.h"
+
+namespace brave_sync {
+
+using bookmarks::BookmarkModel;
+using bookmarks::BookmarkNode;
+
+// This is a wrapper for the BookmarkModelObserver if we need only
+// BookmarkModelLoaded notification
+class BookmarkModelLoadedOnlyObserver
```

Comment:
I'm not understanding the point of this class. This is even more verbose than just implementing empty methods on BraveProfileSyncServiceImpl

Reply:
@bridiver  because just 1 of 8 methods is used and 7 of 8 are not used in `BraveProfileSyncServiceImpl`. I had the idea to avoid increasing of `brave_profile_sync_service_impl.h` and to make it easier read when there is one method which is used then 7 which are not

Reply:
I just don't think it's worth it to add another class just for that and also it makes things harder to understand imo. I had to reference `BookmarkModelLoadedOnlyObserver` because it wasn't obvious why it didn't subclass `BookmarkModelObserver`.

Reply:
Introducing new classes just to stub methods only increases the load on people who are trying to read the code because it's one more new thing they have to learn. If it was more generally useful then there might be a better argument for it, but that isn't the case here since it's only used in one class.

Reply:
this was fixed at https://github.com/brave/brave-core/pull/4588/files#diff-4718e5fe0385d7ab2103a3e905d5fb09R77